modules/caddyhttp/reverseproxy/httptransport.go

```diff
@@ -146,7 +149,59 @@ func (h *HTTPTransport) NewTransport(ctx caddy.Context) (*http.Transport, error)
 	dialer := &net.Dialer{
 		Timeout:       time.Duration(h.DialTimeout),
 		FallbackDelay: time.Duration(h.FallbackDelay),
-		// TODO: Resolver
+	}
+
+	if h.Resolver != nil {
+		// Setup the boostrap logic
+		if len(h.Resolver.Bootstrap) != 0 {
+			for _, v := range h.Resolver.Bootstrap {
+				addr, err := caddy.ParseNetworkAddress(v)
+				if err != nil {
+					return nil, err
+				}
+				if addr.PortRangeSize() != 1 {
+					return nil, fmt.Errorf("bootstrap resolver address must have exactly one address; cannot call %v", addr)
+				}
+				if bootstrapIP := net.ParseIP(addr.Host); bootstrapIP != nil {
+					h.Resolver.bootstrapAddrs = append(h.Resolver.bootstrapAddrs, addr)
+				} else {
+					return nil, fmt.Errorf("bootstrap resolver bootstrap address must be an IP address to avoid mutually recursive lookup; cannot use %v", addr)
+				}
+			}
+			d := &net.Dialer{
+				Timeout:       time.Duration(h.DialTimeout),
+				FallbackDelay: time.Duration(h.FallbackDelay),
+			}
+			net.DefaultResolver = &net.Resolver{
+				PreferGo: true,
+				Dial: func(stdctx context.Context, _, _ string) (net.Conn, error) {
+					addr := h.Resolver.bootstrapAddrs[weakrand.Intn(len(h.Resolver.bootstrapAddrs))]
+					return d.DialContext(ctx, addr.Network, addr.JoinHostPort(0))
+				},
+			}
```

Comment:
Thought of a better way as soon as I clicked the `Create Pull Request` button! I don't need to replace the `net.DefaultResolver` 🤦‍♂️  I can just set this `*net.Resolver` that calls the bootstrap address as the resolver in the `net.Dialer.Resolver` field of the dialer between the lines 194-197.